[beefsausagee] iP

alvislowyh opened the pull request:
# XiaoBai
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

XiaoBai frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/beefsausagee/ip/releases/tag/A-Release).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

alvislowyh committed:
Implement changes for Level-0

alvislowyh committed:
Level 1

alvislowyh committed:
Level-2

kwuunnn commented on src/main/java/Duke.java:
```diff
@@ -1,8 +1,51 @@
+    }
+
+    /**
+     * Displays a greeting message
```
I think the sentence should end with a fullstop. I noticed the same issue in several other places too.

kwuunnn commented on src/main/java/Duke.java:
```diff
@@ -1,8 +1,51 @@
+    /**
+     * Displays a greeting message
+     * 
+     * @param botName Name of the bot
```
Likewise, I think each parameter description should end with a fullstop. I noticed the same issue in several other places too.

kwuunnn commented on src/main/java/Duke.java:
```diff
@@ -1,8 +1,51 @@
+     * 
+     * @param botName Name of the bot
+     */
+    public static void greet(String botName) {
```
Not sure there's a need to make the botName a parameter since it's probably only going to be used during greeting. Maybe just hardcode the botName in the greeting message? Then there would be no need to declare the botName variable in main().

kwuunnn commented on src/main/java/Duke.java:
```diff
@@ -1,8 +1,51 @@
+    private static final String LINE = "___________________________________________________________\n";
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
```
Interesting how you used BufferedReader instead of Scanner. Is there any advantages of using it?

kwuunnn commented on src/main/java/Duke.java:
```diff
@@ -1,8 +1,51 @@
+     * 
+     * @param input User input
+     */
+    public static void repeatFunction(String input) {
```
Not sure I like the name 'repeat'. Maybe echo would have been a more appropriate name?

kwuunnn reviewed (commented):
LGTM. Just a few nits to fix.

alvislowyh committed:
Level 3. Mark As Done

alvislowyh committed:
Level 3. Mark As Done

alvislowyh committed:
Level 4. ToDos, Events, Deadlines

alvislowyh committed:
Level 5. Handle Errors

alvislowyh committed:
Level 5. Handle Errors

alvislowyh committed:
Level 6. Delete

alvislowyh committed:
Level 6. Delete

alvislowyh committed:
Text UI Testing

xuelinglow commented on src/main/java/tempCodeRunnerFile.java:
```diff
@@ -0,0 +1,138 @@
+     * @param task        Task input
+     * @param listOfTasks List of all tasks
+     */
+    public static void repeatFunction(Task task, ArrayList<? extends Task> listOfTasks) {
```
Perhaps it would be better to use a more descriptive function name to make it more clear what the function does 
e.g notifyTaskAdded

xuelinglow reviewed (commented):
So far looks pretty good in terms of code quality guidelines.

alvislowyh committed:
Level-7

alvislowyh committed:
Level 8

alvislowyh committed:
Level 8

alvislowyh committed:
Implemented changes for MoreOOP

alvislowyh committed:
Implemented changes for packages

alvislowyh committed:
Merge branch 'add-gradle-support'

alvislowyh committed:
Added JUnit Test for Deadline class and Event class

alvislowyh committed:
Package the App as a JAR file

alvislowyh committed:
Add JavaDoc comments

alvislowyh committed:
Tweak the code to comply with a coding standard

alvislowyh committed:
Implement changes for Level-9

alvislowyh committed:
Merge changes from branch-Level-9

alvislowyh committed:
Merge branch 'branch-Level-9'

alvislowyh committed:
Merge changes from branch-A-JavaDoc

alvislowyh committed:
Implement changes to Duke and Task

alvislowyh committed:
Implement changes to Duke

alvislowyh committed:
Add comments to Ui and Duke

alvislowyh committed:
Merge branch 'branch-A-CodingStandard'

alvislowyh committed:
Implement changes for Level 10

alvislowyh committed:
Implement changes for Level-10

alvislowyh committed:
Add XiaoBai and User image

alvislowyh committed:
Change Duke to XiaoBai

alvislowyh committed:
Implement changes: Duke to XiaoBai

alvislowyh committed:
Implement changes for code quality

There are some methods in UI that did not follow the naming standard

Changing the name imporves code quality

alvislowyh committed:
Add assertions to improve input validation and error handling

This commit introduces assertions to enhance input validation and error handling in the XiaoBai application. Assertions are used throughout the codebase to ensure that critical assumptions about the input data are met, helping to catch potential bugs and improve overall code reliability.

- Assertions have been added to the `getResponse` method in the `XiaoBai` class to check for empty task details in various user commands, including TODO, DEADLINE, EVENT, MARK, UNMARK, DELETE, and FIND.
- When empty task details are detected, assertions throw an AssertionError, providing clear feedback to developers during testing and debugging.
- Additionally, appropriate error handling has been implemented to catch AssertionError and handle it gracefully, preventing unexpected termination of the application.

This enhancement enhances the robustness of the application and contributes to a more stable and reliable user experience. It lays the groundwork for more thorough input validation and error handling strategies in future iterations of the XiaoBai application.

alvislowyh committed:
Merge pull request #2 from beefsausagee/branch-A-Assertions

Add assertions to improve input validation and error handling

alvislowyh committed:
Add Javadoc comments to all classes in the project

This commit adds Javadoc comments to all public methods and classes in the project to improve code documentation and readability. Each class and method now has a comprehensive Javadoc comment explaining its purpose, parameters, return values (if applicable), and any exceptions thrown.

The following changes were made:
- Added Javadoc comments to all public methods and classes in the project.
- Updated the comments to follow standard conventions and provide clear explanations of each component's functionality.
- Ensured consistency in commenting style and format throughout the project.
- Reviewed and refined existing comments to ensure accuracy and completeness.

These Javadoc comments will enhance code maintainability and facilitate better understanding for developers working on the project. They serve as comprehensive documentation for the project's API, making it easier for users to utilize and extend the functionality of the application.

alvislowyh committed:
Merge pull request #3 from beefsausagee/branch-A-CodeQuality

Add Javadoc comments to all classes in the project

alvislowyh committed:
Removed txt file

alvislowyh committed:
Implement changes to tasks

alvislowyh committed:
Added extension C-Sort

alvislowyh committed:
Merge pull request #4 from beefsausagee/branch-C-Sort

Added extension C-Sort

alvislowyh committed:
Implemented separate methods for processing todo, deadline, event, mark, unmark, delete, and find commands

alvislowyh committed:
Merge branch 'master' of https://github.com/beefsausagee/ip

alvislowyh committed:
Implement changes to Storage, UI, XiaoBai

alvislowyh committed:
Implement changes to text outcome to include personality for XiaoBai

alvislowyh committed:
Implemented changes to MainWindow.fxml to change design of GUI window
Added Ui.png into docs

alvislowyh committed:
Implemented changes to README.md

alvislowyh committed:
Implemented changes for assertions

alvislowyh committed:
Implemented changes to README.md

alvislowyh committed:
Implemented changes to README.md

alvislowyh committed:
Implemented changes to runtest

alvislowyh committed:
Implemented changes to Storage

alvislowyh committed:
Implemented changes to UG

alvislowyh committed:
Add additional date and time formats

alvislowyh committed:
Implement changes to parser

alvislowyh committed:
Implement changes to Storage description

alvislowyh committed:
Implement changes to Ui for personality

alvislowyh committed:
Implement changes to XiaoBaiException to only display error message

minghowC reviewed (commented):
Well implemented. LGTM, just one nit to fix. Perhaps magic number could be better explained with named constants in the case of line 32? 
Pardon my PR at this juncture of time.
